tjpcov/nmt_tools.py

```diff
@@ -0,0 +1,438 @@
+
+    return cl_cp
+
+
+def get_workspace(f1, f2, m1, m2, bins, outdir, **kwards):
+    """
+    Return the workspace of the fields f1, f2
+
+    Parameters:
+    -----------
+        f1 (NmtField):  Field 1
+        f2 (NmtField):  Field 2
+        m1 (str): Mask name assotiated to the field 1
+        m2 (str): Mask name assotiated to the field 2
+        bins (NmtBin):  NmtBin instance
+        outdir (str): Path to the output folder where to store the
+        workspace. If None, no file will be saved.
+        mask_names (dict): Dictionary with tracer names as key and maks names
+        as values.
+        **kwards:  Extra arguments to pass to
+        `w.compute_coupling_matrix`. In addition, if recompute=True is
+        passed, the cw will be recomputed even if found in the disk.
+
+    Returns:
+    --------
+        w:  NmtCovarianceWorkspace of the fields f1, f2, f3, f4
+
+    """
+    if outdir is not None:
+        fname = os.path.join(outdir, f'w__{m1}__{m2}.fits')
+        isfile = os.path.isfile(fname)
+
+        # The workspace of m1 x m2 and m2 x m1 is the same.
+        fname2 = os.path.join(outdir, f'w__{m2}__{m1}.fits')
+        isfile2 = os.path.isfile(fname2)
+    else:
+        fname = isfile = fname2 = isfile2 = None
+
+    w = nmt.NmtWorkspace()
+    if 'recompute' in kwards:
+        recompute = kwards.pop('recompute')
+    else:
+        recompute = False
+
+    if recompute or ((not isfile) and (not isfile2)):
+        w.compute_coupling_matrix(f1, f2, bins, **kwards)
+        if fname:
+            w.write_to(fname)
+        if isfile2:
+            # Remove the other to avoid later confusions
+            os.remove(fname2)
+    elif isfile:
+        w.read_from(fname)
+    else:
+        w.read_from(fname2)
+
+    return w
+
+
+def get_covariance_workspace(f1, f2, f3, f4, m1, m2, m3, m4, outdir, **kwards):
+    """
+    Return the covariance workspace of the fields f1, f2, f3, f4
+
+    Parameters:
+    -----------
+        f1 (NmtField):  Field 1
+        f2 (NmtField):  Field 2
+        f3 (NmtField):  Field 3
+        f4 (NmtField):  Field 4
+        m1 (str): Mask name assotiated to the field 1
+        m2 (str): Mask name assotiated to the field 2
+        m3 (str): Mask name assotiated to the field 3
+        m4 (str): Mask name assotiated to the field 4
+        outdir (str): Path to the output folder where to store the
+        workspace. If None, no file will be saved.
+        **kwards:  Extra arguments to pass to
+        `nmt.compute_coupling_coefficients`. In addition, if recompute=True is
+        passed, the cw will be recomputed even if found in the disk.
+
+    Returns:
+    --------
+        cw:  NmtCovarianceWorkspace of the fields f1, f2, f3, f4
+
+    """
+    # Any other symmetry?
+    combinations = [(m1, m2, m3, m4), (m2, m1, m3, m4), (m1, m2, m4, m3),
+                    (m2, m1, m4, m3), (m3, m4, m1, m2), (m4, m3, m1, m2),
+                    (m3, m4, m2, m1), (m4, m3, m2, m1)]
+
```

Comment:
All the different combinations of masks should have the same covariance workspace, so we check that there is not a covariance workspace of any of those combinations already in the disk. In case none of them exist (i.e. `not True in isfiles`), then it compute it and save it with the first combination of mask names.

Reply:
Sounds good. 

A minor comment: I will recommend using np.any(isfiles) instead of not True in isfiles .

Reply:
Done